File: php/features.php
<?php
declare(strict_types=1);

// RestApiBeispiele SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RestApiBeispieleFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RestApiBeispieleBaseFeature();
            case "test":
                return new RestApiBeispieleTestFeature();
            default:
                return new RestApiBeispieleBaseFeature();
        }
    }
}

File: py/features.py
# RestApiBeispiele SDK feature factory

from feature.base_feature import RestApiBeispieleBaseFeature
from feature.test_feature import RestApiBeispieleTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RestApiBeispieleBaseFeature(),
        "test": lambda: RestApiBeispieleTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
